components/Form/Form.tsx

```diff
@@ -1,18 +1,28 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {View, Text} from 'react-native';
 import Input from '../Input/Input';
 import {Controller, useForm} from 'react-hook-form';
 import Button from '../Button/Button';
-import {FormButton, FormField} from '../../types/CommonTypes';
+import {Data, FormButton, FormField} from '../../types/CommonTypes';
 
 import style from './style';
 import globalStyle from '../../assets/styles/globalStyle';
 
 const Form: React.FC<{
   fields: FormField[];
   buttons: FormButton[];
-}> = ({fields, buttons}) => {
-  const {handleSubmit, control} = useForm();
+  formData?: Data;
```

Comment:
`formData` + `useEffect` is the same as `defaultValues` yes?

If that's the case you should use the already provided `defaultValues` from the library. Keep in mind that it will run only on the first render, if that doesn't help, you should use `setValue` instead, not `reset`. `reset` is supposed to be used to reset the form to a previous state not to update it with a new one.